File: test/basic.test.js
test('basic addition test', () => {
  expect(2 + 2).toBe(4);
});

test('string test', () => {
  expect('hello'.toUpperCase()).toBe('HELLO');
});

test('list test', () => {
  const list = [1, 2, 3];
  expect(list.length).toBe(3);
});

test('test with output', () => {
  console.log('some test output');
  expect(true).toBe(true);
});

File: test/basic_test.rs
#[cfg(test)]
mod tests {
    #[test]
    fn test_addition() {
        assert_eq!(2 + 2, 4);
    }

    #[test]
    fn test_string() {
        let s = "hello";
        assert_eq!(s.to_uppercase(), "HELLO");
    }

    #[test]
    fn test_vector() {
        let v = vec![1, 2, 3];
        assert_eq!(v.len(), 3);
    }

    #[test]
    fn test_with_output() {
        println!("some test output");
        assert!(true);
    }
}